src/test/java/io/jenkins/plugins/analysis/core/graphs/HealthSeriesBuilderTest.java

```diff
@@ -0,0 +1,444 @@
+  private final static int TOO_OLD_LIMIT = 8;
+  private final static int BUILD_COUNT_LIMIT = 3;
+  private final static long MILLISECONDS_OF_HOUR = 60 * 60 * 1000;
+  private final static long MILLISECONDS_OF_MIN = 60 * 1000;
+  private final static AnalysisBuild TOO_OLD_BUILD = stubAnalysisBuild(1, TOO_OLD_LIMIT+1,0,0);
+  //</editor-fold>
+
+  //<editor-fold desc="Day before Yesterday">
+  private final static AnalysisBuild BUILD_FROM_THREE_DAYS_AGO_MIDNIGHT = stubAnalysisBuild(2, 3,0,0);
+  private final static AnalysisBuild BUILD_FROM_THREE_DAYS_AGO_MIDDAY = stubAnalysisBuild(3, 3,12,0);
+  private final static AnalysisBuild BUILD_FROM_THREE_DAYS_AGO_NIGHT = stubAnalysisBuild(4, 3,23,59);
+  //</editor-fold>
+  //<editor-fold desc="Day before Yesterday">
+  private final static AnalysisBuild BUILD_FROM_TWO_DAYS_AGO_MIDNIGHT = stubAnalysisBuild(5, 2,0,0 );
+  private final static AnalysisBuild BUILD_FROM_TWO_DAYS_AGO_MIDDAY= stubAnalysisBuild(6, 2,12,0);
+  private final static AnalysisBuild BUILD_FROM_TWO_DAYS_AGO_AFTERNOON = stubAnalysisBuild(7, 2,15,0);
+  private final static AnalysisBuild BUILD_FROM_TWO_DAYS_AGO_NIGHT = stubAnalysisBuild(8, 2,23,59);
+  //</editor-fold>
+  //<editor-fold desc="Yesterday">
+  private final static AnalysisBuild BUILD_FROM_ONE_DAY_AGO_MIDNIGHT = stubAnalysisBuild(9, 1,0,0);
+  private final static AnalysisBuild BUILD_FROM_ONE_DAY_AGO_MORNING = stubAnalysisBuild(11, 1,6,0);
+  private final static AnalysisBuild BUILD_FROM_ONE_DAY_AGO_MIDDAY = stubAnalysisBuild(12, 1,12,0);
+  private final static AnalysisBuild BUILD_FROM_ONE_DAY_AGO_AFTERNOON = stubAnalysisBuild(13, 1,15,0);
+  private final static AnalysisBuild BUILD_FROM_ONE_DAY_AGO_NIGHT = stubAnalysisBuild(14, 1,23,59);
+  //</editor-fold>
+
+
+
+
+
+
+  //<editor-fold desc="useBuildDateAsDomain=false">
+  @Test
+  void createEmptyDataSetUseBuildDateNotAsDomain(){
+    CategoryDataset result = new HealthSeriesBuilderTestBuilder()
+            .createCategoryDataSet();
+    assertSize(result, 0,0);
+  }
+  @Test
+  void dataSetPerBuildWithOneBuild(){
+    CategoryDataset result = new HealthSeriesBuilderTestBuilder()
+            .setHealth(1)
+            .setUnhealthy(1)
+            .setResultTime(null) // switch time off.
+            .isDayCountDefined(true)
+            .addAnalysisRun(42, BUILD_FROM_ONE_DAY_AGO_MIDNIGHT)
+            .createCategoryDataSet();
+
+    assertSize(result, 1,1);
+    assertValuesInVerticalOrder(result, 42);
+    assertColumnKeysAsNumberOnlyBuildLabel(result, BUILD_FROM_ONE_DAY_AGO_MIDNIGHT);
+  }
+  @Test
+  void dataSetWithTwoBuildFromSameDay(){
+    CategoryDataset result = new HealthSeriesBuilderTestBuilder()
+            .isDayCountDefined(true)
+            .addAnalysisRun(1, BUILD_FROM_ONE_DAY_AGO_MIDDAY)
+            .addAnalysisRun(6, BUILD_FROM_ONE_DAY_AGO_AFTERNOON)
+            .createCategoryDataSet();
+
+    assertSize(result, 2,3);
+    assertValuesInVerticalOrder(result, 1, 0, 0, 1,4,1);
```

Comment:
Insgesamt (nicht in diesem Testfall) sicherstellen, dass die beiden Grenzen von allen Seiten geprüft werden.